src/visitors/sympy_solver_visitor.cpp

```diff
@@ -424,47 +398,25 @@ void SympySolverVisitor::visit_diff_eq_expression(ast::DiffEqExpression& node) {
     check_expr_statements_in_same_block();
 
     const auto node_as_nmodl = to_nmodl_for_sympy(node);
-    const auto locals = py::dict("equation_string"_a = node_as_nmodl,
-                                 "dt_var"_a = codegen::naming::NTHREAD_DT_VARIABLE,
-                                 "vars"_a = vars,
-                                 "use_pade_approx"_a = use_pade_approx,
-                                 "function_calls"_a = function_calls);
-
+    auto diffeq_solver =
+        nmodl::pybind_wrappers::EmbeddedPythonLoader::get_instance().api()->create_des_executor();
+    diffeq_solver->node_as_nmodl = node_as_nmodl;
+    diffeq_solver->dt_var = codegen::naming::NTHREAD_DT_VARIABLE;
+    diffeq_solver->vars = vars;
+    diffeq_solver->use_pade_approx = use_pade_approx;
+    diffeq_solver->function_calls = function_calls;
+    diffeq_solver->method = solve_method;
+    (*diffeq_solver)();
     if (solve_method == codegen::naming::EULER_METHOD) {
-        logger->debug("SympySolverVisitor :: EULER - solving: {}", node_as_nmodl);
         // replace x' = f(x) differential equation
         // with forwards Euler timestep:
         // x = x + f(x) * dt
-        py::exec(R"(
-                from nmodl.ode import forwards_euler2c
-                exception_message = ""
-                try:
-                    solution = forwards_euler2c(equation_string, dt_var, vars, function_calls)
-                except Exception as e:
-                    # if we fail, fail silently and return empty string
-                    solution = ""
-                    exception_message = str(e)
-            )",
-                 py::globals(),
-                 locals);
+        logger->debug("SympySolverVisitor :: EULER - solving: {}", node_as_nmodl);
     } else if (solve_method == codegen::naming::CNEXP_METHOD) {
         // replace x' = f(x) differential equation
         // with analytic solution for x(t+dt) in terms of x(t)
         // x = ...
         logger->debug("SympySolverVisitor :: CNEXP - solving: {}", node_as_nmodl);
-        py::exec(R"(
-                from nmodl.ode import integrate2c
-                exception_message = ""
-                try:
-                    solution = integrate2c(equation_string, dt_var, vars,
-                                           use_pade_approx)
-                except Exception as e:
-                    # if we fail, fail silently and return empty string
-                    solution = ""
-                    exception_message = str(e)
-            )",
-                 py::globals(),
-                 locals);
     } else {
         // for other solver methods: just collect the ODEs & return
```

Comment:
I think not? at least I did not modify this branch... can you elaborate?

Reply:
If I am reading correctly - initially `py::exec(...)` was only called for two `if` statements (EULER and CNEXP). In the last `else` block we just collect the ODEs. Hence thought  `(*diffeq_solver)();` shouldn't be called outside (i.e. in case of `else` statement).

Reply:
Ah, but then you see that inside `diffeq_solver` we have the same branches and there the `else` is also left empty.

Reply:
Ah ok..it's bit confusing because rest of code is inside and last block outside. But its ok.